File: Crank/paiza5.py
n = int(input())

for i in range(n):
    a,b,c = map(str, input().split())
    d,f = a.split(":")

    hour = int(d) + int(b) 
    minute = int(c) + int(f) 
    
    if minute >= 60:
        minute -= 60
        hour += 1
    
    if hour >= 24:
        hour -= 24
    
    if len(str(hour)) == 1:
        hour = "0" + str(hour)
    if len(str(minute)) == 1:
        minute = "0" + str(minute)
        
    print(str(hour) + ":" + str(minute))

File: morinocontest9/paiza5.swift
let input = readLine()!.split(separator: " ").map { Int($0)! }
let n = input[0]
let k = input[1]
let a = readLine()!.split(separator: " ").map { Int($0)! }

var dp = Array(repeating: Array(repeating: 0, count: k+1), count: n+1)
dp[0][0] = 1

for i in 1...n { 
    for j in 0...k { 
        dp[i][j] = dp[i-1][j] 
        if j >= a[i-1] { 
        dp[i][j] += dp[i-1][j-a[i-1]] 
            } 
    } 
}
print(dp[n][k])